stablehlo/reference/Ops.cpp

```diff
@@ -78,6 +78,14 @@ Tensor evalBroadcastInDimOp(const Tensor &operand,
   return result;
 }
 
+SmallVector<Tensor> evalCaseOp(const Tensor &index,
+                               MutableArrayRef<Region> branches, Scope &scope) {
```

Comment:
Can we use `RegionRange` instead?

https://github.com/llvm/llvm-project/blob/c2e248c6aea6396dbfe3bd3ed08ce587b5d63283/mlir/include/mlir/IR/Region.h#L321
 
The subsequent code will look like. 
```
return eval(*branches[idx], {}, &scope);
```

Use of `MutatbleArrayRef` gives the impression that we are going to change the regions in someway. `RegionRange` avoids that.